reactioncommerce/reaction
# Fix #3875: card validator

Resolves #3875  
Impact: **major**  
Type: **bugfix**

## Issue
Card validator regex was checking for number in the range of 14-16 though 12-19 is what it should be checking for.

## Solution
Fixed the regex and moved the validator regex to `helper` for reuse.

## Testing
1. Enable Authorize.net payment.
2. Enter a card with just 12 numbers(424242424242)
3. Observe that you don't get a `Match failed` error.
4. Try entering a number with less than those digit. Observe you get the error that the card should have atleast 12 digits.
